lib/nodes/inline-comment.js

```diff
@@ -7,7 +10,23 @@ module.exports = {
 
       while (token) {
         if (/\n/.test(token[1])) {
-          this.tokenizer.back(token);
+          // If there are quotes, fix tokenizer creating one token from start quote to end quote
+          if (/['"].*\n/.test(token[1])) {
+            // Add string before newline to inline comment token
+            bits.push(token[1].substring(0, token[1].indexOf('\n')));
+
+            // Get remaining input and retokenize
+            let remainingInput = token[1].substring(token[1].indexOf('\n'));
+            remainingInput += this.input.css.valueOf().substring(this.tokenizer.position());
+
+            // Replace tokenizer to retokenize the rest of the string
+            const newInput = new Input(remainingInput);
+            this.input = newInput;
```

Comment:
This is something interesting, I do not know the entirety of the side effects of doing this. I am replacing the input file with the input file given to the new tokenizer when I replace it. Presumably this could have some impact on error messages generated afterwards (but replacing the tokenizer probably has similar impact, so I don't know if this matters).

Reply:
it's a novel approach, I'll give you that.

there are some stylistic changes I'd like to see happen, but I'll make those after the fact.